osu.Game/Screens/Play/KeyCounter.cs

```diff
@@ -1,151 +1,99 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-#nullable disable
-
-using osu.Framework.Allocation;
+using System;
+using osu.Framework.Bindables;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
-using osu.Framework.Graphics.Sprites;
-using osu.Framework.Graphics.Textures;
-using osu.Game.Graphics;
-using osu.Game.Graphics.Sprites;
-using osuTK;
-using osuTK.Graphics;
 
 namespace osu.Game.Screens.Play
 {
     public abstract partial class KeyCounter : Container
     {
-        private Sprite buttonSprite;
-        private Sprite glowSprite;
-        private Container textLayer;
-        private SpriteText countSpriteText;
+        public readonly InputTrigger Trigger;
 
-        public bool IsCounting { get; set; } = true;
-        private int countPresses;
+        private readonly Container content;
 
-        public int CountPresses
+        protected override Container<Drawable> Content => content;
+
+        private readonly Bindable<int> countPresses = new BindableInt
         {
-            get => countPresses;
-            private set
-            {
-                if (countPresses != value)
-                {
-                    countPresses = value;
-                    countSpriteText.Text = value.ToString(@"#,0");
-                }
-            }
-        }
+            MinValue = 0
+        };
 
-        private bool isLit;
+        public bool IsCounting { get; set; } = true;
 
-        public bool IsLit
+        public IBindable<int> CountPresses => countPresses;
+
+        protected KeyCounter(InputTrigger trigger)
         {
-            get => isLit;
-            protected set
+            InternalChildren = new Drawable[]
             {
-                if (isLit != value)
+                content = new Container
                 {
-                    isLit = value;
-                    updateGlowSprite(value);
-                }
-            }
+                    RelativeSizeAxes = Axes.Both
+                },
+                Trigger = trigger,
+            };
+
+            Trigger.OnActivate += Activate;
+            Trigger.OnDeactivate += Deactivate;
+
+            Name = trigger.Name;
         }
 
+        protected Bindable<bool> IsActive = new BindableBool();
+
         public void Increment()
         {
             if (!IsCounting)
                 return;
 
-            CountPresses++;
+            countPresses.Value++;
         }
 
         public void Decrement()
         {
             if (!IsCounting)
                 return;
 
-            CountPresses--;
+            countPresses.Value--;
         }
```

Comment:
Not sure these should be public now. Only callers are the protected event flows.